src/os_posix/os_mtx_cond.c

```diff
@@ -165,7 +166,8 @@ __wt_cond_signal(WT_SESSION_IMPL *session, WT_CONDVAR *cond)
      * Fast path if we are in (or can enter), a state where the next waiter will return immediately
      * as already signaled.
      */
-    if (cond->waiters == -1 || (cond->waiters == 0 && __wt_atomic_casi32(&cond->waiters, 0, -1)))
+    cond_waiters = __wt_atomic_loadi32(&cond->waiters);
```

Comment:
The previous iteration of this code looks wrong in a few ways, I think this is an improvement.

Wouldn't the fact that we are trying to swap in `0` as the `old` value here mean that we don't actually need to check `cond_waiter == 0`? I think we can leave it as is but this is a bit funky.

Reply:
I think the cond_waiter check is a short circuit so we can avoid making an unnecessary (and more expensive?) cas call. Agreed this can stay as is